pkg/util/cloudfoundry/cloudfoundry_test.go

```diff
@@ -16,31 +16,51 @@ func TestHostAliasDisable(t *testing.T) {
 	mockConfig.Set("cloud_foundry", false)
 	mockConfig.Set("bosh_id", "ID_CF")
 
-	alias, err := GetHostAlias()
+	aliases, err := GetHostAliases()
 	assert.Nil(t, err)
-	assert.Empty(t, alias)
+	assert.Nil(t, aliases)
 }
 
 func TestHostAlias(t *testing.T) {
 	mockConfig := config.Mock()
 
 	mockConfig.Set("cloud_foundry", true)
 	mockConfig.Set("bosh_id", "ID_CF")
+	mockConfig.Set("cf_os_hostname_aliasing", false)
 
-	alias, err := GetHostAlias()
+	aliases, err := GetHostAliases()
 	assert.Nil(t, err)
-	assert.Equal(t, "ID_CF", alias)
+	assert.Equal(t, []string{"ID_CF"}, aliases)
+
+	mockConfig.Set("cf_os_hostname_aliasing", true)
+	aliases, err = GetHostAliases()
+	assert.Nil(t, err)
+
+	hostname, _ := os.Hostname()
+	fqdn := util.Fqdn(hostname)
+	if hostname == fqdn {
```

Comment:
Nit: I see why you're doing this, but this is the sort of test that would not test everything and might hide a dormant bug depending on the machine where it runs. 

What we're currently doing in that `if...else` is very basic, so I'm suggesting we change this as a "best practice" sort of thing. If you could try to mock `util.Fqdn` output to force both cases, that would be best.

Reply:
🙈 
I tried mocking in go once and said to myself, never again. I'll look into it again 👍